samples/todo-android/app/src/main/java/com/squareup/sample/mainactivity/ToDoActivity.kt

```diff
@@ -0,0 +1,68 @@
+package com.squareup.sample.mainactivity
+
+import android.os.Bundle
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.squareup.sample.container.overviewdetail.OverviewDetailContainer
+import com.squareup.sample.todo.TodoListsAppWorkflow
+import com.squareup.workflow1.SimpleLoggingWorkflowInterceptor
+import com.squareup.workflow1.diagnostic.tracing.TracingWorkflowInterceptor
+import com.squareup.workflow1.ui.ViewRegistry
+import com.squareup.workflow1.ui.WorkflowLayout
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import com.squareup.workflow1.ui.backstack.BackStackContainer
+import com.squareup.workflow1.ui.renderWorkflowIn
+import kotlinx.coroutines.flow.StateFlow
+import java.io.File
+
+@OptIn(WorkflowUiExperimentalApi::class)
+class ToDoActivity : AppCompatActivity() {
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+
+    val model: ToDoModel by viewModels()
+
+    setContentView(
+      WorkflowLayout(this).apply {
+        start(model.ensureWorkflow(getExternalFilesDir(null)!!), viewRegistry)
+      }
+    )
+  }
+
+  private companion object {
+    val viewRegistry =
+      ViewRegistry(
+        TodoEditorLayoutRunner,
+        TodoListsViewFactory,
+        OverviewDetailContainer,
+        BackStackContainer
+      )
+  }
+}
+
+class ToDoModel(private val savedState: SavedStateHandle) : ViewModel() {
+  private lateinit var renderings: StateFlow<Any>
+
+  @OptIn(WorkflowUiExperimentalApi::class)
+  fun ensureWorkflow(externalFilesDir: File): StateFlow<Any> {
+    if (!::renderings.isInitialized) {
```

Comment:
I think `isInitialized` is almost always a code smell. Just make the property nullable.